proposals/0020-hlsl-202x-202y.md

```diff
@@ -68,8 +75,60 @@ have workarounds and may pose a barrier to adoption.
 
 ### HLSL 202y
 
-HLSL 202y shall contain features that expand HLSL. This is all other outstanding
-feature proposals.
+HLSL 202y is an opportunity to address some frequently requested features that
+we can deliver more quickly in Clang than in DXC. Some of the features in HLSL
+202y require significant changes to core language behaviors and will likely
+cause source compatibility issues.
+
+Two big areas of investment in HLSL 202y are overload resolution and object
+initialization. While the two areas are intertwined, we will discuss them mostly
+separately.
+
+#### Overload Resolution
+
+In Clang we've adopted overload resolution behavior more similar to C++. This
+causes some source incompatibilities as cited in the [clang
+docs](https://clang.llvm.org/docs/HLSL/ExpectedDifferences.html). Taking these
+behavior changes into HLSL makes HLSL's overload best-match algorithm more
+familiar and will produce ambiguous resolution errors instead of potentially
+choosing the wrong overload when ambiguity exists in the source.
+
+Adopting C++ overload best match and set selection rules implements proposal
+[0007 `const` non-static member
+functions](https://github.com/microsoft/hlsl-specs/blob/main/proposals/0007-const-instance-methods.md).
+
+Adopting C++ language rules for the contexts in which overload resolution occurs
+implements proposal [0008 non-member operator
+overloading](https://github.com/microsoft/hlsl-specs/blob/main/proposals/0008-non-member-operator-overloading.md).
+
+#### Object Initialization
+
+Supporting [C++
+constructors](https://github.com/microsoft/hlsl-specs/issues/303) requires
+revisiting HLSL's approach where all objects support aggregate initialization
+(braced initializer lists), which makes it connected to proposal [0005 strict
+initializer
+lists](https://github.com/microsoft/hlsl-specs/blob/main/proposals/0005-strict-initializer-lists.md).
+
+HLSL 2021 & 202x have special rules for aggregate initializers such that the
+source-provided structure of braced initializer lists is ignored and an
+element-wise initialization is performed. This conflicts with objects having
+constructors, destructors, or user-provided conversion functions.
+
+HLSL 202y adopts support for constructors and destructors, and C++'s rules for
+aggregate initialization.
+
+#### Other Mostly Free Features
+
+Other C++ features have made their way in via Clang's implementation preserving
+C++ constructs, but not all will be standardized in HLSL 202y. A few notable
+features that should make the cut are:

```

Comment:
I think there's some nuance here that might need expanding on for someone not immersed in thinking about language standards,  What's the implication of a feature being available and seeming to work in Clang in 202y mode but not actually being standardized in 202y?

Or is the plan to disable these features that we don't manage to get standardized?

Reply:
I think most of them we'll disable outright. Some (like lambdas) we might test and find they work well enough that maybe we can just leave them in, although at that point we should probably decide what language version they're part of.

I'll work on the wording around this.